ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/prepackaged_dataflow_applications.md

```diff
@@ -0,0 +1,63 @@
+# Prepackaged DF: one way to hook up multiple components
+
+## Use case
+You have already known that OCI Data Flow is a powerful service for running Spark applications, but one problem is that it will allow you to upload only one application (code) file per application. This may force you to white all the logics in one file, which is bad for mid-large scale applications where modularization is desired.
```

Comment:
Good catch. Fixed

Reply:
actually let's reword as follows: 

"You might be aware that OCI Data Flow is a powerful service for running Spark applications. For Python applications, Data Flow supports reading from an archive file in addition to the main application file. This allows modularization which is required for development of mid- to large- sized applications"

Otherwise it seems we are critical of our own product

Reply:
Good catch! However, Archive is not only for Python applications. So I change it to the following:

```
You might be aware that OCI Data Flow is a powerful service for running Spark applications, generated by uploading a single code application in Java, Python SQL and Scala. For better coding pattern, Data Flow supports modularization by reading from an archive file as an library in addition to the main application file. This helps customers to work with the development of mid- to large- sized applications.
```